lit_gpt/adapter.py

```diff
@@ -253,7 +253,7 @@ def forward(
             ay = self.scaled_dot_product_attention(q, ak, av, amask)
             y = y + self.gating_factor * ay
```

Comment:
I am just eyeballing the code (sorry for not running on a pc prior to commenting), but this looks a bit off.

`ay` should have a shape something similar to `(B, nh, T, hs)`. Like it's describe in annotations in [lit-llama](https://github.com/Lightning-AI/lit-llama/blob/main/lit_llama/adapter.py#L164).
And gating factor now is `(1, 1, nh, 1)`. 
It looks like it should fail, shouldn't it?

Reply:
P.S. In not so distant future I'll add annotation from `lit-llama` with some necessary updates.

Reply:
The transpose is being done directly in `scaled_dot_product_attention` now

Reply:
😊 
Feel so embarrassed. I see `scaled_dot_product_attention` and immidiately think that it's from `torch.nn.functional` while it's a class method with the same name.
All I had to do is to simply scroll down ... :)